lib/orocos/process.rb

```diff
@@ -1171,14 +1183,32 @@ def self.wait_running(process, timeout = nil, name_service = Orocos::CORBA.name_
                     block.call(process)
                 else
                     # Get any task name from that specific deployment, and check we
-                    # can access it. If there is none
+                    # can access it either through the ior pipe or through the CORBA name
+                    # service.
+                    if process.alive?
+                        begin
+                            ior_message += ior_read.read_nonblock 4096
+                        rescue IO::WaitReadable
+                        end
+
+                        if ior_read.eof?
+                            if ior_message_valid? process, ior_message
```

Comment:
Make ior_message an internal instance variable (@ior_message) instead of passing it as argument. wait_running will be called repeatedly, during which it has to "save" ior_message.